### DISCUSSION TITLE
vue3+vite Uncaught TypeError: _ctx.$t is not a function

### DISCUSSION THREAD

**Discussion by alireza0sfr** [intlify/bundle-tools#122] [Q&A]
hi there, I want to publish my vue3+vite package to npm but after publishing, I encountered "Uncaught TypeError: _ctx.$t is not a function" in a test project and my package is not working, any suggestions... ?

PS: I'm using options api

here is my vite.configs.js
```
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import vueI18n from '@intlify/vite-plugin-vue-i18n'

// https://vitejs.dev/config/
const path = require("path")
export default defineConfig({
  build: {
    lib: {
      entry: path.resolve(__dirname, 'src/install.ts'),
      name: 'vcp',
      fileName: (format) => `vcp.${format}.ts`
    },
    rollupOptions: {
      external: ['vue'],
      output: {
        exports: 'named',
        globals: {
          vue: 'Vue',
          vcp: 'Vcp'
        }
      }
    },
  },
  plugins: [
    vue(),
    vueI18n({
      include: path.resolve(__dirname, 'src/assets/translations.ts'),
      compositionOnly: false,
    })
  ],
  server: {
    port: 8080
  },
  resolve: {
    dedupe: ['vue'],
    alias: {
      "~": path.resolve(__dirname, "./src"),
      "@": path.resolve(__dirname, "./src"),
    },
  },
})
```

**Comment by kazupon** [comment]
You need to external `vue-i18n`, and vue-i18n should install in your test project.

**Reply by alireza0sfr** [reply]
hey thanks for your response, but the main problem is that users that download my package will have to deal with vue-i18n and translations file but optimally it should be handled internally, and user only has to install my package and everything should work out of the box, if I external vue-i18n user needs to install vue-i18n by itself and register it and use my translations file which he/she doesn't have!

**Reply by alireza0sfr** [reply]
I want to do something like this but it doesn't work on the test project I get "messageCompiler not found " error

install.ts(my entry): 
```
// @ts-ignore
import componentRegisterer from './plugins/components.ts'
// @ts-ignore
import mixins from './plugins/mixins.ts'
// @ts-ignore
import i18n from './plugins/i18n.ts'

export default {
  install: (app: any, options: any): void => {
    app.mixin(mixins)
    componentRegisterer(app)
    app.use(i18n)
  }
}
```

**Reply by alireza0sfr** [reply]
@kazupon HELP WATNTED!

**Reply by kazupon** [reply]
Why are you using the plugin mechanism to install? 🤔 
In any case, we can't solve this problem without reproduced code.

**Reply by alireza0sfr** [reply]
how should I install it then? 
I can't provide reproducible code now but i work on it and add it to this discussion soon

**Reply by alireza0sfr** [reply]
@kazupon Hey there, I managed to finish things up and be able to prepare a reproducible code for you, the only thing left is the internationalization system

to review my error for you:
I want to be auto-injected and use my own package's translations and the user that downloads my package doesn't have to deal with translations at all
here is the repo: https://github.com/alireza0sfr/vue-client-print
and here is the docs: https://alireza0sfr.github.io/vue-client-print/
thanks in advance

**Reply by kazupon** [reply]
Thank you for your reproducing repo!
I've sent PR to it!
https://github.com/alireza0sfr/vue-client-print/pull/2

**Reply by alireza0sfr** [reply]
Thanks for the quick response and fix.
by the way, do you have any plans to add ts support to @intlify/vite-plugin-vue-i18n ?

**Reply by alireza0sfr** [reply]
@kazupon hey there, there is a problem with your commits when I published the package to the local registry and used the artifact translations will replace the main project's translations, and only my package translations worked.

update: 
I need to change the scope to local therefore library doesn't change the main project's i18n instance
I'm using options API and I tested "sync", "useScope" but couldn't fix the issue

**Reply by artemklishch** [reply]
Hello!
I am in process of installing settings on project that is going to be written on Nuxt3. 
I haven't found any instructions about it (there is a warning on the web-site that browser testing API is still in stage of development), installed default Nuxt 2 project, took test configuration from there and put it into my project with replacement versions in some places.  And installed VueTestUtils library to try if it will work. 
Tests started to run but I get the similar issue -  'TypeError: _ctx.$t is not a function'
Also I get the error 'ReferenceError: definePageMeta is not defined'
Looks as if tests don't see global variables (I use Composition API: <script setup lang='ts'>...</script>)

I also get the error 'Property 'isVueInstance' does not exist on type 'VueWrapper<any, any>'.' - looks like TypeScript error.

Could you please advise me where can I apply to get explanations or instruction to install unit tests correctly for the Nuxt 3?
Thank you!
